### DISCUSSION TITLE
Please help me to resolve this error in prolif.

### DISCUSSION THREAD

**Discussion by Shivani8878** [chemosim-lab/ProLIF#199] [Q&A]
Hi, I am using md simulation trajectory to get protein ligand interactions from prolif. but i am facing an error please see the attached image.
![prolif error](https://github.com/chemosim-lab/ProLIF/assets/106540989/d886d17d-c6f4-4eb7-a2a6-2b3c24cf0bb9)

**Comment by cbouy** [comment]
The atom types are not recognized by MDAnalysis, try swapping the types with the names before creating the selections for the ligand and protein :

`u.atoms.types = u.atoms.names`

**Reply by Shivani8878** [reply]
I tried that but its not working.
I used here.Then tried same error.
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/43b77c21-fa1c-49eb-aef3-efcac4253700)
Then, I added here. but same error.
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/01a5d21e-0e2e-4432-a059-5622522f4500)

**Reply by Shivani8878** [reply]
This time error has changed to 
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/21c5e9f1-2137-4702-a720-652f1cbe9390)

**Reply by cbouy** [reply]
Try the following instead then:

```
from MDAnalysis.topology.guessers import guess_types 

u.atoms.types = guess_types(u.atoms.names) 
```

**Reply by Shivani8878** [reply]
A new error has come now
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/63c439f9-2a1d-4aca-87a6-7d55b594ffdb)

**Reply by cbouy** [reply]
Try adding `u.atoms.elements = u.atoms.types` and if that doesn't work please share the content of elements, types and atom names

**Reply by Shivani8878** [reply]
I have got this output. But the structure of ligand JE0 is not showing completely. and P3- is coming Don't know why.
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/3b56f44c-1928-454d-ab32-50e10b6c60b1)


And in the interaction diagram JE0 structure is showing perfectly but this extra P3- is coming.
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/e46c4f7d-78e4-4224-bb24-7bcc58499831)

I wanted to see protein interaction with three ligand molecules (JE0, ATP, MG+2)

**Reply by Shivani8878** [reply]
I can provide you with the gro file.
[Uploading md_0_1.txt…](https://github.com/chemosim-lab/ProLIF/discussions/199)

**Reply by cbouy** [reply]
Mdanalysis had to guess bonds as your topology does not contain them explicitly, it mostly worked except for JE0 where it ends up fragmented because of a missing bond. 

You can customize the van der Waals radii used to guess the bonds by calling `ligand_selection.guess_bonds(vdwradii={"P": 1.80})` and try increasing that 1.80 value until JE0 is not fragmented anymore (and it should be displayed correctly by `display_residues`).

Note that you'll also have to explicitely call guess_bonds for the protein selection as well (but you shouldn't need to provide custom vdw radii there)

**Reply by Shivani8878** [reply]
Another error:
![image](https://github.com/chemosim-lab/ProLIF/assets/106540989/357aa44c-9642-418f-8a81-849c36f39a2b)

**Reply by cbouy** [reply]
Try smaller increments for the vdwradii, something like 2.0, 2.2... Etc.
The error you get means some atoms are now ending with more bonds than they should (i.e. the opposite of what your problem was) so if you end up in that situation I'd suggest contacting the MDAnalysis people instead to see if they have suggestions other than guessing bonds

**Comment by rwxayheee** [comment]
Hi @Shivani8878 Maybe MDA does not guess well for some of the atom names you originally had (`HGPAM1`? `LP1`?). Can you post your inputs or share with @cbouy ?

**Reply by Shivani8878** [reply]
I can provide you with the gro file. 
[Uploading md_0_1.txt…]()
